bot/exts/info/tags.py

```diff
@@ -172,125 +246,163 @@ async def search_tag_content(self, ctx: Context, *, keywords: str) -> None:
+            return tag.embed
 
-            if cooldown_conditions:
-                return True
-            return False
-
-        if _command_on_cooldown(tag_name):
-            time_elapsed = time.time() - self.tag_cooldowns[tag_name]["time"]
-            time_left = constants.Cooldowns.tags - time_elapsed
-            log.info(
-                f"{ctx.author} tried to get the '{tag_name}' tag, but the tag is on cooldown. "
-                f"Cooldown ends in {time_left:.1f} seconds."
+        else:
+            if not filtered_tags:
+                return None
+            suggested_tags_text = "\n".join(
+                f"**\N{RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK}** {identifier}"
+                for identifier, tag in filtered_tags
+                if not tag.on_cooldown_in(ctx.channel)
+            )
+            return Embed(
+                title="Did you mean ...",
+                description=suggested_tags_text
             )
-            return True
-
-        if tag_name is not None:
-            temp_founds = self._get_tag(tag_name)
-
-            founds = []
-
-            for found_tag in temp_founds:
-                if self.check_accessibility(ctx.author, found_tag):
-                    founds.append(found_tag)
-
-            if len(founds) == 1:
-                tag = founds[0]
-                if ctx.channel.id not in TEST_CHANNELS:
-                    self.tag_cooldowns[tag_name] = {
-                        "time": time.time(),
-                        "channel": ctx.channel.id
-                    }
 
-                self.bot.stats.incr(f"tags.usages.{tag['title'].replace('-', '_')}")
+    async def list_all_tags(self, ctx: Context) -> None:
+        """Send a paginator with all loaded tags accessible by `ctx.author`, groups first, and alphabetically sorted."""
+        def tag_sort_key(tag_item: tuple[TagIdentifier, tag]) -> str:
+            ident = tag_item[0]
+            if ident.group is None:
+                # Max codepoint character to force tags without a group to the end
+                group = chr(0x10ffff)
+            else:
+                group = ident.group
+            return group+ident.name
+
+        result_lines = []
+        current_group = object()
+        group_accessible = True
+
+        for identifier, tag in sorted(self._tags.items(), key=tag_sort_key):
+
+            if identifier.group != current_group:
+                if not group_accessible:
+                    # Remove group separator line if no tags in the previous group were accessible by the user.
+                    result_lines.pop()
+                # A new group began, add a separator with the group name.
+                if identifier.group is not None:
+                    group_accessible = False
+                    result_lines.append(f"\n\N{BULLET} **{identifier.group}**")
+                else:
+                    result_lines.append("\n\N{BULLET}")
+                current_group = identifier.group
```

Comment:
You could move this like next to the comment about "A new group began".

Reply:
Moved the assignment to the comment and changed the uses to use the name directly instead of identifier.group in 5cbb7d20713c0a7027c2e0909ddcdb6e02dbddc2